src/utils/util/resolver-pipeline/oscal-profile-resolve-merge.xsl

```diff
@@ -25,34 +28,43 @@
             templates not this one. -->
     </xsl:template>
 
-    <xsl:template match="catalog" priority="2">
+    <!-- If there is no selection and no merge (i.e., we don't reach one of
+        the higher-priority templates that match catalog), pass the catalog
+        through unchanged. -->
+    <xsl:template match="catalog" priority="2" as="element(catalog)">
         <xsl:copy-of select="."/>
     </xsl:template>
 
-    <xsl:template match="catalog[exists(selection)]" priority="10">
+    <!-- If there is a selection but neither merge/as-is nor merge/custom,
+        use flat structuring. -->
+    <xsl:template match="catalog[exists(selection)]" priority="10" as="element(catalog)">
         <catalog>
             <xsl:apply-templates select="@*"/>
             <xsl:apply-templates select="metadata"/>
             <xsl:apply-templates select="selection"/>
             <xsl:apply-templates select="modify"/>
-            <xsl:for-each-group select="back-matter/* | selection/back-matter/*" group-by="true()">
+            <xsl:where-populated>
                 <back-matter>
-                    <xsl:copy-of select="current-group()"/>
+                    <xsl:sequence select="back-matter/* | selection/back-matter/*"/>
                 </back-matter>
-            </xsl:for-each-group>
+            </xsl:where-populated>
         </catalog>
     </xsl:template>
 
-    <xsl:template priority="12" match="catalog[merge/as-is=('true','1')]">
+    <!-- If there is a merge/as-is directive, we go down that branch.
+        If there is also a merge/custom directive, we apply the
+        higher-priority template instead of this one. -->
+    <xsl:template match="catalog[merge/as-is=$true-content]" priority="12" as="element(catalog)">
+        <xsl:call-template name="detect-multiple-structuring-directives"/>
```

Comment:
Adding detection here and in custom merge template because the spec says, "If more than one appears, processors MUST generate an error and cease processing."